packages/suite/src/support/messages.ts

```diff
@@ -1,6 +1,24 @@
 import { defineMessages } from 'react-intl';
 
-export default defineMessages({
+type MessageDescriptor<K extends string> = {
+    // Must correspond to the property name.
+    id: K;
+    // Default text in English. This value is only used directly if the corresponding key is missing from the JSON translation files.
+    defaultMessage: string;
+    // Not integrated into Crowdin so not really used.
+    description?: string;
+    // Must be set to true for programmatically constructed keys. Otherwise, the keys will be deleted by the list-duplicates script.
+    dynamic?: boolean;
+};
+
+/**
+Checks whether id corresponds to the property name. Otherwise, text is not translated. Returns the same value that was passed in.
+ */
+const defineMessagesWithTypeCheck = <Keys extends string>(messages: {
+    [K in Keys]: MessageDescriptor<K>;
```

Comment:
Nit: a string union should IMHO be named with singular rather than plural, which is for Array type.
I think it's a convention, but I don't have any source to back it up :smile: 
```suggestion
const defineMessagesWithTypeCheck = <TranslationKey extends string>(messages: {
    [K in TranslationKey]: MessageDescriptor<K>;
```

Reply:
I changed `Keys` => `Key`.